rllib/core/models/configs.py

```diff
@@ -194,81 +222,154 @@ class CNNEncoderConfig(ModelConfig):
     """
 
     input_dims: Union[List[int], Tuple[int]] = None
-    filter_specifiers: List[List[Union[int, List[int]]]] = field(
+    cnn_filter_specifiers: List[List[Union[int, List[int]]]] = field(
         default_factory=lambda: [[16, [4, 4], 2], [32, [4, 4], 2], [64, [8, 8], 2]]
     )
-    filter_layer_activation: str = "relu"
+    cnn_activation: str = "relu"
+    cnn_use_layernorm: bool = False
+    output_dims: Union[List[int], Tuple[int]] = None
     output_activation: str = "linear"
+    use_bias: bool = True
 
-    @_framework_implemented(tf2=False)
+    def _validate(self, framework: str = "torch"):
+        if len(self.input_dims) != 3:
+            raise ValueError(
+                f"`input_dims` ({self.input_dims}) of CNNEncoderConfig must be a 3D "
+                "tensor (image) with the dimensions meaning: width x height x "
+                "channels, e.g. `[64, 64, 3]`!"
+            )
+        if self.output_dims is None or len(self.output_dims) != 1:
+            raise ValueError(
+                f"`output_dims` ({self.output_dims}) of CNNEncoderConfig must be "
+                "a 1D tensor describing the (after flattening) output dimension of "
+                "the CNN, e.g. `[256]`!"
+            )
+
+    @_framework_implemented()
     def build(self, framework: str = "torch") -> Model:
-        # Activation functions in TF are lower case
-        self.output_activation = _convert_to_lower_case_if_tf(
-            self.output_activation, framework
-        )
-        self.filter_layer_activation = _convert_to_lower_case_if_tf(
-            self.filter_layer_activation, framework
-        )
+        self._validate()
 
         if framework == "torch":
             from ray.rllib.core.models.torch.encoder import TorchCNNEncoder
 
             return TorchCNNEncoder(self)
 
+        elif framework == "tf2":
+            from ray.rllib.core.models.tf.encoder import TfCNNEncoder
+
+            return TfCNNEncoder(self)
+
 
 @ExperimentalAPI
 @dataclass
-class MLPEncoderConfig(MLPHeadConfig):
+class MLPEncoderConfig(_MLPConfig):
     """Configuration for an MLP that acts as an encoder.
 
-    Although it inherits from MLPHeadConfig, it does not output an MLPHead.
-    This inheritance is solely to unify the configuration options between MLPEncoders
-    and MLPHeads.
+    See _MLPConfig for usage details.
+    Note that MLPEncoders may specify `output_dims` as None in case only the hidden
+    layers with their specified activations should be part of the resulting network.
 
-    See ModelConfig for usage details.
+    Example:
+    .. code-block:: python
+        # Configuration:
+        config = MLPEncoderConfig(
+            input_dims=[4],  # must be 1D tensor
+            hidden_layer_dims=[16],
+            hidden_layer_activation="relu",
+            hidden_layer_use_layernorm=False,
+            output_dims=None,  # maybe None or a 1D tensor
+        )
+        model = config.build()
```

Comment:
build should always take in a framework!

Reply:
great catch! fixed.